src/lib/krb5/asn.1/asn1_encode.c

```diff
@@ -26,105 +26,100 @@
 
 #include "asn1_encode.h"
 
+struct asn1buf_st {
+    uint8_t *ptr;               /* Position, moving backwards; may be NULL */
+    size_t count;               /* Count of bytes written so far */
+};
+
 /**** Functions for encoding primitive types ****/
 
-krb5_error_code
-k5_asn1_encode_bool(asn1buf *buf, intmax_t val, size_t *len_out)
+/* Insert one byte into buf going backwards. */
+static inline void
+insert_byte(asn1buf *buf, uint8_t o)
+{
+    if (buf->ptr != NULL)
+        *--buf->ptr = o;
```

Comment:
Could we do this as a two-liner instead?

Reply:
Sure:

```C
    if (buf->ptr != NULL) {
        buf->ptr--;
        *buf->ptr = o;
    }
    buf->count++;
```